basket/news/views.py

```diff
@@ -833,7 +833,9 @@ def lookup_user(request):
             return invalid_email_response()
 
     try:
-        user_data = get_user_data(token=token, email=email, get_fxa=get_fxa)
+        user_data = get_user_data(
+            token=token, email=email, get_fxa=get_fxa, masked=not email
```

Comment:
I think we should enable returning the unmasked email when they use an API key regardless of whether they've requested via email or token. 

We also need to do this to the `user()` view since that is the one that bedrock uses to fetch user data.

Reply:
Reading the `lookup_user` code, an API key is only validated when requested with the `email` param. Also, the `user` view doesn't support user lookup by email, just token, so it makes sense to always return masked emails.

Unless you're saying you want a change in the API to allow user lookups with token + API key?

Reply:
Maybe I am. I'm thinking that this will add to the current situation. Currently to use `lookup-user` with email requires an API key because that's privileged info. And now I think we should add to that such that if you send a valid API key along with a request to either of these endpoints you get the full email address back, but otherwise you only get a redacted one. Does that make sense?

Reply:
Yeah, makes sense. So API key plus either token or email replies with unmasked email. I can start working towards that in this PR.